### PR TITLE
brownfield: convert hostName to hostnames in brownfield

### PR DESCRIPTION
**Pull request by akshaysngupta**
## Checklist
- [x] The title of the PR is clear and informative
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

## Description

This PR updates the brownfield workflow to use hostnames field from the HTTP listener in App Gateway for determining the blacklists.
I have tried to handle both hostName and HostNames in the HTTP listener to maintain backend compatibility.

### CODE REVIEW

File: pkg/brownfield/listeners.go [hunk 1]
```diff
@@ -49,7 +62,10 @@ func MergeListeners(listenerBuckets ...[]n.ApplicationGatewayHTTPListener) []n.A
 				Protocol: listener.Protocol,
 			}
 			if listener.HostName != nil {
-				listenerConfig.HostName = *listener.HostName
+				listenerConfig.setHostNames([]string{*listener.HostName})
+			}
+			if listener.Hostnames != nil && len(*listener.Hostnames) > 0 {
```
Comment (3quanfeng) [inline pkg/brownfield/listeners.go:67]: the second setHostNames() always overrides the one for the HostName case above?

Comment (akshaysngupta) [reply pkg/brownfield/listeners.go:67]: only one of the above conditions can be true at a time.

File: pkg/brownfield/listeners.go [hunk 2]
```diff
@@ -34,12 +40,19 @@ func (er ExistingResources) GetBlacklistedListeners() ([]n.ApplicationGatewayHTT
...
-	HostName                string
+	HostNames               [MaxAllowedHostnames]string
 	Protocol                n.ApplicationGatewayProtocol
 	FrontendPortID          string
 	FrontendIPConfiguration string
 }
 
+func (config *uniqueListenerConfig) setHostNames(hostnames []string) {
+	hostnameCount := int(math.Min(float64(len(hostnames)), float64(MaxAllowedHostnames)))
+	for i := 0; i < hostnameCount; i++ {
```
Comment (3quanfeng) [inline pkg/brownfield/listeners.go:51]: a warning if the host count max out?

Comment (akshaysngupta) [reply pkg/brownfield/listeners.go:51]: this should handled else where.

**Pull request merged**